AP_HAL_ChibiOS: add hwdef for LongBowF405WING

HiRCBro opened the pull request:
hi
    lefeirc PR new board with F405

magicrub commented on libraries/AP_HAL_ChibiOS/hwdef/LongBowF405WING/hwdef.dat:
```diff
@@ -0,0 +1,196 @@
+define AP_BATTMON_SMBUS_ENABLE 0
+
+include ../include/minimize_fpv_osd.inc
+include ../include/save_some_flash.inc
```
This should be removed. It's not compiling

magicrub commented on libraries/AP_HAL_ChibiOS/hwdef/LongBowF405WING/hwdef.dat:
```diff
@@ -0,0 +1,196 @@
+MCU STM32F4xx STM32F405xx
+
+# board ID for firmware load
+APJ_BOARD_ID 1422
```
This needs to be added to Tools/AP_Bootloader/board_types.txt

magicrub reviewed (dismissed):
Looks pretty good, thanks! This needs a bootloader. You can compile it by running:
`./Tools/scripts/build_bootloaders.py LongBowF405WING`

magicrub commented:
The commits should also be squashed

tridge commented:
@HiRCBro I've fixed up the commits in your PR

Hwurzburg commented:
In order to review I will need pinout/board image.....also, any UART intended for RC operation needs DMA  so UART2 in alt configuration will not support CRSF/ELRS and perhaps even SRXL2 ptotocols....so those need to be on another UART...but no others have full DMA either...so RC protocols using bidir serial will have an issue with current definition I beleive

Hwurzburg commented:
the pinnout of each jst-gh connector is missing, labeled solder pads are clear, but what signals are in each JST-GH are not indicated....

HiRCBro commented:
> In order to review I will need pinout/board image.....also, any UART intended for RC operation needs DMA so UART2 in alt configuration will not support CRSF/ELRS and perhaps even SRXL2 ptotocols....so those need to be on another UART...but no others have full DMA either...so RC protocols using bidir serial will have an issue with current definition I beleive

SBUS connect to UART2 with an inverter, if you use SBUS then UART2 cant work. if you use CRSF you cant connect to SBUS you need connect to UART2 or other UART

Hwurzburg commented:
UART2 and all other UARTs do not have full DMA required for CRSF/ELRS for sure...so you cant make those RC systems work reliably with this hwdef

HiRCBro commented:
> UART2 and all other UARTs do not have full DMA required for CRSF/ELRS for sure...so you cant make those RC systems work reliably with this hwdef
--------------------
So i need change
PA2 USART2_TX USART2 NODMA
PA3 USART2_RX USART2 NODMA ALT(1)
to:
PA2 USART2_TX USART2 
PA3 USART2_RX USART2ALT(1)
？

Hwurzburg commented:
unfortunately dma does not get assigned to alternate function pins....so that wont do it....I dont know of a way to do it on a uart whose primary pin function is RCINT for RC .....best to get one of the other UARTs DMA enabled....UART6 unfortunately was not assigned a DMA channel in the resolver....maybe @Tridge can make a suggestion, or you cna play with NODMA on some UARTS and see what the resolver produces for the others....you have to look at the hwdef.h file in the build directory after to waf configure for you board

HiRCBro commented:
> se primary pin function is RCINT

i am confused, i test uart2 with CRSF is OK, do you mean, CRSF must work with a full DMA uart, i check hwdef.h only uart1 is full DMA with RX and TX.

HiRCBro commented:
> > se primary pin function is RCINT
> 
> i am confused, i test uart2 with CRSF is OK, do you mean, CRSF must work with a full DMA uart, i check hwdef.h only uart1 is full DMA with RX and TX.

if uart2 not suitbale for CRSF, we can use UART1, it is full DMA with RX and TX

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/LongBowF405WING/Readme.md:
```diff
@@ -0,0 +1,100 @@
+ - BATT_MONITOR 4
+ - BATT_VOLT_PIN 10
+ - BATT_CURR_PIN 11
+ - BATT_VOLT_MULT 11.5
```
```suggestion
 - BATT_VOLT_MULT 11.0
```
per hwdef

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/LongBowF405WING/Readme.md:
```diff
@@ -0,0 +1,100 @@
+## VTX power control
+
+GPIO 81 controls the VTX BEC output to pins marked "9V". Setting this GPIO high removes voltage supply to pins.
+
```
```suggestion
# Camera Switch

GPIO 82 controls which camera input (CC1 or C2 is applied to the internal OSD. A RELAY function can be enabled to control the switching.

# Buzzer 

An active buzzer output is provided and is controlled as GPIO 80
```

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/LongBowF405WING/Readme.md:
```diff
@@ -0,0 +1,100 @@
+# LongBowF405WING Flight Controller
+
+The LongBowF405WING is a flight controller produced by [lefei rc](http://www.lefeirc.com/).
+
```
insert link to images here
also need an image of the PDB since it has external connections (probably ESCpwer out and VBAT in)

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/LongBowF405WING/hwdef.dat:
```diff
@@ -0,0 +1,180 @@
+define HAL_BATT_VOLT_SCALE 11.0 # matched to PDB board
+define HAL_BATT_CURR_SCALE 50  # matched to PDB board
+
+# analog rssi pin
```
I cannt find that pinned out anywhere

HiRCBro replied:
> I cannt find that pinned out anywhere
This hardware has not been introduced, it is only reserved.

Hwurzburg replied:
so this is just a strawman at this point....please let me/us know if and when its real hardware and we will review it then

HiRCBro replied:
> so this is just a strawman at this point....please let me/us know if and when its real hardware and we will review it then

HI， The current hardware does not have an interface, if it is not suitable, you can delete it first

Hwurzburg replied:
we can leave them....for a future bd iteration perhaps....it does not hurt anything to have them defined,but not accessible

Hwurzburg commented:
> > se primary pin function is RCINT
> 
> i am confused, i test uart2 with CRSF is OK, do you mean, CRSF must work with a full DMA uart, i check hwdef.h only uart1 is full DMA with RX and TX.

its a matter of RELIABILITY.....CRSF and to a lesser extent ELRS will work without DMA, but not reliably...

HiRCBro commented:
> > > se primary pin function is RCINT
> > 
> > 
> > i am confused, i test uart2 with CRSF is OK, do you mean, CRSF must work with a full DMA uart, i check hwdef.h only uart1 is full DMA with RX and TX.
> 
> its a matter of RELIABILITY.....CRSF and to a lesser extent ELRS will work without DMA, but not reliably...

Understood, DMA can reduce CPU  load and improve data processing capabilities

HiRCBro commented:
> > > se primary pin function is RCINT
> > 
> > 
> > i am confused, i test uart2 with CRSF is OK, do you mean, CRSF must work with a full DMA uart, i check hwdef.h only uart1 is full DMA with RX and TX.
> 
> its a matter of RELIABILITY.....CRSF and to a lesser extent ELRS will work without DMA, but not reliably...

HI
    When can this be merged  into the master，thankyou.

Hwurzburg commented:
once the open review comments are addressed, the commits squashed and split into two commits one for Tools and one for hwdef it can be posted for devcall review and probable merge after that call....
if you wish, I can fix the readme recommended changes and squash and split commits for you....let me know

HiRCBro commented:
> once the open review comments are addressed, the commits squashed and split into two commits one for Tools and one for hwdef it can be posted for devcall review and probable merge after that call....
> if you wish, I can fix the readme recommended changes and squash and split commits for you....let me know

Please help me complete it， thank you

Hwurzburg reviewed (approved):
I have applied the changes, rebased, squashed, and library split....since I never got the board (delivered to wrong address I suppose) please retest the firmware on your board...once confirmed I will mark it for devcall (next one is Monday night GMT)

HiRCBro commented:
> I have applied the changes, rebased, squashed, and library split....since I never got the board (delivered to wrong address I suppose) please retest the firmware on your board...once confirmed I will mark it for devcall (next one is Monday night GMT)



> I have applied the changes, rebased, squashed, and library split....since I never got the board (delivered to wrong address I suppose) please retest the firmware on your board...once confirmed I will mark it for devcall (next one is Monday night GMT)

HI 
why  LongBowF405WING still cant get from https://firmware.ardupilot.org/Plane/stable-4.5.7/

Hwurzburg commented:
because backports must be requested...I have marked for backport...but I am not sure it will be since 4.6 is about to beta release soon..

Hwurzburg commented:
4.6 will be automatic...its backports that are not...but IF another 4.5 release occurs, it should be added now

HiRCBro commented:
Ok. thank you

---- Replied Message ----
| From | Henry ***@***.***> |
| Date | 10/15/2024 20:43 |
| To | ArduPilot/ardupilot ***@***.***> |
| Cc | HiRCBro ***@***.***>,
Mention ***@***.***> |
| Subject | Re: [ArduPilot/ardupilot] AP_HAL_ChibiOS: add hwdef for LongBowF405WING (PR #27009) |

4.6 will be automatic...its backports that are not...but IF another 4.5 release occurs, it should be added now

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>
